File: GestionEvenementFront/src/app/_Models/Product.ts
import {Resource} from "@angular/compiler-cli/src/ngtsc/metadata";
import {Ordre} from "./Ordre";

export class Product {
  productId: string;
  nameProduct:string;
  price:number ;
  stock:number;
  currentStock:number;
  ordres:Ordre[];
  quantity:number;
}

File: GestionEvenementBack/src/main/java/tn/eesprit/gestionevenementback/Entities/Product.java
package tn.eesprit.gestionevenementback.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import javax.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class Product implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer productId;
    String nameProduct;
    Float price;
    Integer stock;
    @JsonIgnore
    @OneToMany(mappedBy = "product")
    Set<Ordre> ordres;

}
